From cbc4b5a2742b66027c52632f899edf99602883d2 Mon Sep 17 00:00:00 2001
From: SamuXarick <43006711+SamuXarick@users.noreply.github.com>
Date: Wed, 18 Feb 2026 09:01:18 +0000
Subject: [PATCH 1046/1053] Fix 434163aa: Crashed zeppelin not blocking runway
 (#15281)

Turns out the block flag was in Reset mode instead of Set.
Move the Set block logic earlier to make it repeatedly set for as long as the zeppeliner exists.
---
 src/disaster_vehicle.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/disaster_vehicle.cpp b/src/disaster_vehicle.cpp
index dc7ff65e7e..cbed1ec829 100644
--- a/src/disaster_vehicle.cpp
+++ b/src/disaster_vehicle.cpp
@@ -218,7 +218,7 @@ void DisasterVehicle::UpdatePosition(int x, int y, int z)
  * 0: Zeppeliner initialization has found an airport, go there and crash
  * 1: Create crash and animate falling down for extra dramatic effect
  * 2: Create more smoke and leave debris on ground
- * 2: Clear the runway after some time and remove crashed zeppeliner
+ * 3: Clear the runway after some time and remove crashed zeppeliner
  * If not airport was found, only state 0 is reached until zeppeliner leaves map
  */
 static bool DisasterTick_Zeppeliner(DisasterVehicle *v)
@@ -258,6 +258,10 @@ static bool DisasterTick_Zeppeliner(DisasterVehicle *v)
 		return true;
 	}
 
+	if (IsValidTile(v->tile) && IsAirportTile(v->tile)) {
+		Station::GetByTile(v->tile)->airport.blocks.Set({AirportBlock::Zeppeliner, AirportBlock::RunwayIn});
+	}
+
 	if (v->state > 2) {
 		if (++v->age <= 13320) return true;
 
@@ -299,10 +303,6 @@ static bool DisasterTick_Zeppeliner(DisasterVehicle *v)
 		v->age = CalendarTime::MIN_DATE;
 	}
 
-	if (IsValidTile(v->tile) && IsAirportTile(v->tile)) {
-		Station::GetByTile(v->tile)->airport.blocks.Reset({AirportBlock::Zeppeliner, AirportBlock::RunwayIn});
-	}
-
 	return true;
 }
 
-- 
2.55.0

